lms/djangoapps/mobile_api/social_facebook/courses/views.py

```diff
@@ -0,0 +1,80 @@
+"""
+Views for courses info API
+"""
+
+
+from rest_framework import generics, status
+from rest_framework.response import Response
+from courseware.access import is_mobile_available_for_user
+from student.models import CourseEnrollment
+from mobile_api.users.serializers import CourseEnrollmentSerializer
+from openedx.core.djangoapps.user_api.api.profile import preference_info
+from ...utils import mobile_view
+from ..utils import get_friends_from_facebook, get_linked_edx_accounts
+from lms.djangoapps.mobile_api.social_facebook.courses import serializers
+from django.conf import settings
+
+_FACEBOOK_API_VERSION = settings.FACEBOOK_API_VERSION
+
+
+@mobile_view()
+class CoursesWithFriends(generics.ListAPIView):
+    """
+    **Use Case**
+
+        API endpoint for retriving all the courses that a users friends are in.
+        Note that only friends that allow their courses to be shared will be included.
+
+    **Example request**
+
+        GET /api/mobile/v0.5/social/facebook/courses/friends
+
+    **Response Values**
+
+        See UserCourseEnrollmentsList in lms/djangoapps/mobile_api/users for the structure of the response values.
+    """
+    serializer_class = serializers.CoursesWithFriendsSerializer
+
+    def list(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.GET, files=request.FILES)
+        if serializer.is_valid():
+            # Get friends from Facebook
+            data = get_friends_from_facebook(serializer)
+            if type(data) != list and data.status_code == 400:
```

Comment:
hmm.. doesn't the code have to handle other status codes here as well?  even if they are not normally expected?  Otherwise, `if type(data) != list` and `data.status_code` is something else, you will get errors downstream in this method.

I suggest having the code on L45-54 be embedded inside a positive `if` statement so you're sure the result was Ok and you can continue with the rest of this code.  In the `else`, you can return an error Response.